File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        HashMap<Character,Integer> freq = new HashMap<>();
        for(char ch : s.toCharArray()){
            if(!freq.containsKey(ch)){
                // freq[ch] =1;
                freq.put(ch,1);
            }
            else{
                freq.put(ch,freq.get(ch)+ 1);
            }
        
        }
        for(int i =0; i<s.length();i++){
            if(freq.get(s.charAt(i))==1){
                return i;
            }
        }
        return -1;
        }
        
    
}

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        freq ={}
        for name in s:
            if name not in freq:
                freq[name] = 1
            else:
                freq[name] += 1
        
        for name in range(len(s)):
            if freq[s[name]] ==  1:
                return name
        return -1 